### DISCUSSION TITLE
mercadopago.preferences.create()

### DISCUSSION THREAD

**Discussion by MustaCH** [mercadopago/sdk-js#189] [Q&A] [answered]
Hi, I'm getting an error when executing my code, it tells me that mercadopago.preferences.create() is not a function. But everything I have seen has worked that way. Could you help me?

`const express = require("express");
const app = express();
const cors = require("cors");
const path = require("path");

const mercadopago = require("mercadopago");

new mercadopago.MercadoPagoConfig({
  access_token:
    "",
});

app.use(express.urlencoded({ extended: false }));
app.use(express.json());

app.use(express.static(path.join(__dirname, "../client")));
app.use(cors());

app.get("/", function (req, res) {
  const filePath = path.resolve(__dirname, "..", "client", "index.html");
  res.sendFile(filePath);
});

app.post("/create_preference", (req, res) => {
  let preference = {
    items: [
      {
        title: req.body.description,
        unit_price: Number(req.body.price),
        quantity: Number(req.body.quantity),
      },
    ],
    back_urls: {
      success: "http://localhost:5500",
      failure: "http://localhost:5500",
      pending: "",
    },
    auto_return: "approved",
  };

  mercadopago.preferences
    .create(preference)
    .then(function (response) {
      res.json({
        id: response.body.id,
      });
    })
    .catch(function (error) {
      console.log(error);
    });
});

app.get("/feedback", function (req, res) {
  res.json({
    Payment: req.query.payment_id,
    Status: req.query.status,
    MerchantOrder: req.query.merchant_order_id,
  });
});

app.listen(5500, () => {
  console.log("The server is now running on Port 5500");
});
`

I share the complete code so that you can take a look at it

**Comment by gdeandradero** [comment, marked as answer]
Hi how are you?
It is likely that you have installed version 2 of the SDK and are implementing version 1.
The recommended option is to implement version 2, in your example there would be the following changes:

from
```node
const mercadopago = require(“mercadopago”);

new mercadopago.MercadoPagoConfig({
access_token:
“”,
});
```

to
```node
import { MercadoPagoConfig, Preference } from ‘MercadoPago’;

const client = new MercadoPagoConfig({ accessToken: ‘YOUR_AT’ });
```

and
```node
let preference = {
        items: [
            {
                title: req.body.description,
                unit_price: Number(req.body.price),
                quantity: Number(req.body.quantity),
            },
        ],
        back_urls: {
            success: “http://localhost:5500”,
            failure: “http://localhost:5500”,
            pending: “”,
        },
        auto_return: “approved”,
    };

    mercadopago.preferences
        .create(preference)
        .then(function (response) {
            res.json({
                id: response.body.id,
            });
        })
        .catch(function (error) {
            console.log(error);
        });
```

to
```node
const body = {
    items: [
        {
            title: req.body.description,
            unit_price: Number(req.body.price),
            quantity: Number(req.body.quantity)
        },
    ],
    back_urls: {
        success: “http://localhost:5500”,
        failure: “http://localhost:5500”,
        pending: “”
    },
    auto_return: “approved”
};

preference.create({
    ‘body’: body
}).then((result) => console.log(result))
    .catch((error) => console.log(error));
```
